src/components/SuppliersComponent.vue

```diff
@@ -0,0 +1,217 @@
+<template>
+  <div id="supplierPage">
+    <button class="btn btn-secondary" v-on:click="showForm('post')">
+      Add new supplier
+    </button>
+
+    <div class="form-popup p-4" id="popupDiv">
+      <form id="addForm">
```

Comment:
Add @submit.prevent="confirmForm" to the <form> and in the confirm <button> remove the event handler and replace the type for "submit". Also consider adding the required field in the form's input. Look at my previous PR as a reference.